app/src/androidTest/java/org/fossasia/openevent/APITest.java

```diff
@@ -36,4 +42,102 @@ public void failure(RetrofitError error) {
+                fail("API Request Failed");
+                latch.countDown();
+            }
+        });
+        latch.await();
+    }
+
+    public void testEventAPIResponse() throws Exception {
+        APIClient client = new APIClient();
+        final CountDownLatch latch = new CountDownLatch(1);
+        client.getOpenEventAPI().getEvents(new Callback<EventResponseList>(){
+            @Override
+            public void success(EventResponseList eventResponseList, Response response) {
+                assertNotNull(eventResponseList.events);
+                assertTrue(eventResponseList.events.size() > 0);
+                latch.countDown();
+            }
+
+            @Override
+            public void failure(RetrofitError error) {
+                fail("API Request Failed");
+                latch.countDown();
+            }
+        });
+        latch.await();
+    }
+    public void testSessionAPIResponse() throws Exception {
+        APIClient client = new APIClient();
+        final CountDownLatch latch = new CountDownLatch(1);
+        client.getOpenEventAPI().getSessions(new Callback<SessionResponseList> () {
+            @Override
+            public void success(SessionResponseList sessionResponseList, Response response) {
+                assertNotNull(sessionResponseList.sessions);
+                assertTrue(sessionResponseList.sessions.size() > 0);
+                latch.countDown();
+            }
+
+            @Override
+            public void failure(RetrofitError error) {
+                fail("API Request Failed");
+                latch.countDown();
+            }
+        });
+        latch.await();
+    }
+    public void testTrackAPIResponse() throws Exception {
+        APIClient client = new APIClient();
+        final CountDownLatch latch = new CountDownLatch(1);
+        client.getOpenEventAPI().getTracks(new Callback<TrackResponseList>() {
+            @Override
+            public void success(TrackResponseList trackResponseList, Response response) {
+                assertNotNull(trackResponseList.tracks);
+                assertTrue(trackResponseList.tracks.size() > 0);
+                latch.countDown();
+            }
+
+            @Override
+            public void failure(RetrofitError error) {
+                fail("API Request Failed");
+                latch.countDown();
+            }
+        });
+        latch.await();
+    }
+    public void testMicrolocationAPIResponse() throws Exception {
+        APIClient client = new APIClient();
+        final CountDownLatch latch = new CountDownLatch(1);
+        client.getOpenEventAPI().getMicrolocations(new Callback<MicrolocationResponseList>() {
+            @Override
+            public void success(MicrolocationResponseList microlocationResponseList, Response response) {
+                assertNotNull(microlocationResponseList.microlocations);
+                assertTrue(microlocationResponseList.microlocations.size() > 0);
```

Comment:
@creativepsyco This was just a getting started thing. I am doing the DB insertion thing right now, then I'll make tests for that. I couldn't test anything else at the stage where I committed this code. So, yeah I did Copy it.